hermes/probe/create_file_test.go

```diff
@@ -1 +1,201 @@
 package probe
+
+import (
+	"fmt"
+	"testing"
+	"bytes"
+	"context"
+
+	"cloud.google.com/go/storage"
+	"github.com/googleapis/google-cloud-go-testing/storage/stiface"
+)
+
+func TestFileName(t *testing.T) {
+	testUnits := [6]RandomHermesFile{ 
+		{51, 12}, // Invalid ID  
+		{0, 50}, // Invalid ID
+		{3, 100}, // Expected Hermes_03_checksum
+		{12, 100}, // Expected Hermes_12_checksum
+		{3, 0}, // Invalid Size
+		{3, 1001}, // Invalid Size
+	}
+	// Case: testUnits[0] RandomHermesFile{51, 12}
+	want, wantErr := "", "The file ID provided 51 wasn't in the required range [1,50]"
+	got, err := testUnits[0].FileName()
+	gotErr:=err.Error() // error in string form
+	if want!=got {
+		t.Errorf("{51, 12}.FileName() failed expected %v got %v", want, got)
+	}
+	if wantErr!=gotErr {
+		t.Errorf("{51, 12}.FileName() gave wrong error expected %v got %v", wantErr, gotErr)
+	}
+	// Case: testUnits[1] RandomHermesFile{0, 50}
+	want, wantErr = "", fmt.Sprintf("The file ID provided %v wasn't in the required range [1,50]", testUnits[1].ID)
+	got, err = testUnits[1].FileName()
+	gotErr=err.Error() // error in string form
+	if want!=got {
+		t.Errorf("{0, 50}.FileName() failed expected %v got %v", want, got)
+	}
+	if wantErr!=gotErr {
+		t.Errorf("{0, 50}.FileName() gave wrong error expected %v got %v", wantErr, gotErr)
+	}
+	// Case: testUnits[2] RandomHermesFile{3, 100}
+	wantPrefix := "Hermes_03_"
+	got, err = testUnits[2].FileName()
+	if err!=nil {
+		t.Errorf("Unexpected error: %v", err)
+	}
+	gotPrefix := got[0:10]
+	if gotPrefix != wantPrefix {
+		t.Errorf("{3,100}.FileName() failed expected prefix %v got prefix %v", wantPrefix, gotPrefix)
+	} 
+	// Case: testUnits[3] RandomHermesFile{12, 100}
+	wantPrefix = "Hermes_12_"
+	got, err = testUnits[3].FileName()
+	if err!=nil {
+		t.Errorf("Unexpected error: %v", err)
+	}
+	gotPrefix = got[0:10]
+	if gotPrefix != wantPrefix {
+		t.Errorf("{12,100}.FileName() failed expected prefix %v got prefix %v", wantPrefix, gotPrefix)
+	} 
+	// Case: testUnits[4] RandomHermesFile{3, 0}
+	want, wantErr = "", "The file size provided 0 is not a positive number as required"
+	got, err = testUnits[4].FileName()
+	gotErr=err.Error() // error in string form
+	if want!=got {
+		t.Errorf("{3, 0}.FileName() failed expected %v got %v", want, got)
+	}
+	if wantErr!=gotErr {
+		t.Errorf("{3, 0}.FileName() gave wrong error expected %v got %v", wantErr, gotErr)
+	}
+	// Case: testUnits[5] RandomHermesFile{3, 1001}
+	want, wantErr = "", "The file size provided 1001 bytes exceeded the limit 1000 bytes"
+	got, err = testUnits[5].FileName()
+	gotErr=err.Error() // error in string form
+	if want!=got {
+		t.Errorf("{3, 1001}.FileName() failed expected %v got %v", want, got)
+	}
+	if wantErr!=gotErr {
+		t.Errorf("{3, 1001}.FileName() gave wrong error expected %v got %v", wantErr, gotErr)
+	}
+}
+
+func TestChecksum (t *testing.T) {
+	f := RandomHermesFile{11, 100}
+	fTwo := RandomHermesFile{13, 1000}
+	fCopy := RandomHermesFile{11, 100}
```

Comment:
You could use a table driven test [1] to define these instead (and as other have pointed out, it's also applicable in your other test functions.

[1] https://github.com/golang/go/wiki/TableDrivenTests